### PR TITLE
Fix slidingmesh for SA

### PR DESCRIPTION
**Pull request by maxaehle**
## Proposed Changes
`CSlidingInterface::GetDonor_Variable` needs to access the `Primitive` of flow variables and the `Solution` of turbulent variables. Which one is the case, is currently decided by checking `donor_solution->GetnPrimVar() == 2` (true->turbulent, false->flow). If the SA turbulence model is used, the non-overidden "virtual placeholder" `CVariable::GetPrimitive(unsigned long,unsigned long) const` returning zero is incorrectly called.

We propose to fix this by additionally checking for `donor_solution->GetnPrimVar() == 1`.

## Related Work
@maxaehle, @mcmorelli, @BeckettZhou, @EduardoMolina

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [ ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [X] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp [hunk 1]
```diff
@@ -51,20 +51,24 @@ void CSlidingInterface::GetDonor_Variable(CSolver *donor_solution, CGeometry *do
   nDonorVar = donor_solution->GetnPrimVar();
 
   /*---  the number of primitive variables is set to two by default for the turbulent solver ---*/
-  bool turbulent = (nDonorVar == 2) ;
+  bool SA, SST;
+  SA  = (nDonorVar == 1);
```
Comment (WallyMaier) [inline SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp:55]: can you do "bool SA = (nDonor==1) ", etc?

File: SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp [hunk 2]
```diff
@@ -51,20 +51,24 @@ void CSlidingInterface::GetDonor_Variable(CSolver *donor_solution, CGeometry *do
...
-
-    /*---  for turbulent solver retrieve solution and set it as the donor variable ---*/
+  if (SA){
+    /*---  for SA turbulent solver retrieve solution and set it as the donor variable ---*/
+    Donor_Variable[0] = donor_solution->GetNodes()->GetSolution(Point_Donor,0);
+  }
+  else if (SST){
+    /*---  for SST turbulent solver retrieve solution and set it as the donor variable ---*/
     Donor_Variable[0] = donor_solution->GetNodes()->GetSolution(Point_Donor,0);
     Donor_Variable[1] = donor_solution->GetNodes()->GetSolution(Point_Donor,1);
```
Comment (pcarruscag) [inline SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp:65]: Do a loop over nVar...

**Review by pcarruscag — CHANGES_REQUESTED** [review]

**Comment by maxaehle** [comment]
I followed @pcarruscag 's request in commit 12974ab.
BTW it's very confusing that the SA/SST turbulent solvers have `nPrimVar==1,2` but `GetPrimitive` is not overridden and thus returns zero. So another fix to the bug considered in this PR would be to override `GetPrimitive` on the turbulent solvers such that it forwards to `GetSolution`?

File: SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp [hunk 3]
```diff
@@ -47,24 +47,18 @@ void CSlidingInterface::GetDonor_Variable(CSolver *donor_solution, CGeometry *do
...
                                           unsigned long Vertex_Donor, unsigned long Point_Donor) {
 
-  unsigned short iVar, nDonorVar;
-  nDonorVar = donor_solution->GetnPrimVar();
-
-  /*---  the number of primitive variables is set to two by default for the turbulent solver ---*/
-  bool turbulent = (nDonorVar == 2) ;
-
+  const unsigned short nDonorVar = donor_solution->GetnPrimVar();
+  const bool turbulent = (nDonorVar==1) || (nDonorVar==2); // SA or SST
```
Comment (pcarruscag) [inline SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp:51]:
```suggestion
  const auto nDonorVar = donor_solution->GetnPrimVar();
  /// TODO: Replace with approach compatible with any number of variables (e.g. encapsulate in a "solver info" object).
  const bool turbulent = nDonorVar <= 2;
```

**Review by pcarruscag — APPROVED** [review]
Thanks Max, I missed the solution v primitive part of the problem.

**Comment by pcarruscag** [comment]
I don't think forwarding GetPrimitive to GetSolution by default is a good option, a "Slinding state" layer could be appropriate but that does not need to be fixed right now.

File: SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp (hunk 3 above)
Comment (maxaehle) [reply SU2_CFD/src/interfaces/cfd/CSlidingInterface.cpp:51]: Yes, `GetnPrimVar()` seems to be non-zero on all flow and turbulence solvers.

**Comment by maxaehle** [comment]
By "Sliding State layer", do you mean a virtual `CVariable::GetSlidingMeshDonorVariable` that returns 0 by default, but forwards to `GetSolution` on the flow and `GetPrimitive` on the turb variables?

**Comment by pcarruscag** [comment]
I would add that layer to the solver so that the interpolation does not have to bypass it by going straight to the nodes.

**Comment by maxaehle** [comment]
Ok thanks Pedro!

**Pull request merged**